Mathlib/Algebra/MvPolynomial/Basic.lean

```diff
@@ -875,6 +875,32 @@ theorem C_dvd_iff_dvd_coeff (r : R) (φ : MvPolynomial σ R) : C r ∣ φ ↔ 
     IsRegular (∏ n in s, X n : MvPolynomial σ R) :=
   IsRegular.prod fun _ _ ↦ isRegular_X
 
+/-- The finset of nonzero coefficients of a multivariate polynomial. -/
+def coeffs (p : MvPolynomial σ R) : Finset R :=
+  letI := Classical.decEq R
+  Finset.image p.coeff p.support
+
+lemma coeffs_zero : coeffs (0 : MvPolynomial σ R) = ∅ :=
+  rfl
+
+lemma coeffs_one : coeffs (1 : MvPolynomial σ R) ⊆ {1} := by
+  classical
+  simp only [coeffs]
+  rw [Finset.image_subset_iff]
+  simp only [mem_support_iff, ne_eq, mem_singleton, ← C_1, coeff_C]
+  intro n hn
+  simp only [exists_prop, ite_eq_right_iff, not_forall] at hn
+  simp [hn]
+
```

Comment:
```suggestion
classical
  simp_rw [coeffs, Finset.image_subset_iff]
  simp_all [coeff_one]
```

Reply:
Can't you prove the equality rather than just an inclusion?

Reply:
I don't think I can, since I think it is wrong for `R = 0`. In this case the `1` polynomial is `0` and hence has empty coefficient set.

Reply:
Good point, but then you can add a (simp) lemma with the assumption that `R` is not `0` and that in that case there is an equality.

Reply:
Thanks for the suggestion, I have done this in the latest commit.